Sprint 9 solution http api

puffpundb opened the pull request.

puffpundb committed:
Изменен способ хранения истории на связанную таблицу

puffpundb committed:
Update Task.java

Исправил код стайл в 11 и 17 строках. Были лишние пробелы

puffpundb committed:
Update InMemoryHistoryManager.java

Исправлен код стайл. Удалены лишние пробелы

puffpundb committed:
Исправлен метод связывания узлов списка. Ссылка на хвост не обновлялась при добавлении второго элемента.

Исправлено местоположение класса Node. Изначально запутался в ТЗ, подумал, что нужно его создать в том же документе, что и менеджер истории.

puffpundb committed:
Исправлен метод связывания узлов списка. Ссылка на хвост не обновлялась при добавлении второго элемента.

Исправлено местоположение класса Node. Изначально запутался в ТЗ, подумал, что нужно его создать в том же документе, что и менеджер истории.

puffpundb committed:
Удалена лишняя проверка в получении истории.
Исправлена логика добавления нодов в связанный список.
Перенесена проверка ключа в метод remove().
Изменен метод сравнения ссылок на объекты.
Исправлено местоположение класса Node. Переместил его в класс InMemoryHistoryManager

puffpundb committed:
Исправлена логика удаления ноды.
Добавлены тесты удаления при единственной записи в истории и удаление записей в пустой истории.

puffpundb committed:
Добавлен менеджер задач, который работает через файл.
Добавлены тесты.

puffpundb committed:
Добавлен хедер в файл сохранения.
Метод save() делан приватным.
Переопределены все методы менеджера, связанные с действиями над объектами.
Создан утилитарный класс Converter и доработана логика конвертации задач.
Добавлены тесты с записью и загрузкой через файл.
Перенесен метод putTaskToSave() в файловый менеджер.

puffpundb committed:
Добавлено поле header для файла сохранения.
Изменен доступ к полям менеджера.
Переписаны тесты.
Добавлено удаление временных файлов в тесты.

puffpundb committed:
Изменен доступ у putTaskToSave().
Изменено именование константы.
Доработана логика загрузки сабтасков.

puffpundb committed:
Добавлены продолжительность и даты задач.
Добавлен список задач по приоритету(по времени старта).
Изменены циклы на стримы(где было по силам).
Создан абстрактный класс для тестов.
Созданы тесты для нового функционала.

puffpundb committed:
Удалены лишние импорты.
Почищен проект от закомментированного кода.

puffpundb committed:
Поправлен кодстайл в конверторе

puffpundb committed:
Поправлен кодстайл в конверторе

puffpundb committed:
Удалено поле endTime у Task и SubTask.
Изменена логика удаления из сортированного множества.
Изменены методы получения по id из внутренних методов.
Созданы отдельные методы для расчета времени эпика.
Изменена реализация расчета пересечения отрезков.
Добавлено параметризованное поле taskManager и его инициализация в тестах.
Перемещены тесты, которые тестируют общие методы менеджеров.

puffpundb committed:
Merge branch 'sprint_8-solution-time-and-duration' of https://github.com/puffpundb/java-kanban into sprint_8-solution-time-and-duration

# Conflicts:
#	src/managers/InMemoryTaskManager.java

puffpundb committed:
Доисправления по ревью

puffpundb committed:
Файл .iml и директория .idea добавлены в .gitignore.
Логика расчета начала и конца эпика вынесены в один метод.
Доработаны тестовые классы.

puffpundb committed:
Коммит пере слиянием веток

puffpundb committed:
Реализовано API для менеджера

puffpundb committed:
Поправлен код-стайл

Ivanov-s-tmn commented on test/server/HttpMethodDeleteTest.java:
```diff
@@ -0,0 +1,180 @@
+import java.util.List;
+
+public class HttpMethodDeleteTest extends HttpTaskServerTest {
+    @Test
+    public void shouldServerRightDeleteTask() throws IOException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create("http://localhost:8080/tasks/0"))
+                .DELETE()
+                .header("Accept", "application/json")
+                .build();
+        client.send(request, HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8));
+
+        request = HttpRequest.newBuilder()
+                .uri(URI.create("http://localhost:8080/tasks"))
+                .GET()
+                .header("Accept", "application/json")
+                .build();
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8));
+        JsonArray listFromServer = JsonParser.parseString(response.body()).getAsJsonArray();
+        List<Task> listFromJson = GSON.fromJson(listFromServer, new TypeToken<List<Task>>() {
+        }.getType());
+
+        Assertions.assertEquals(0, listFromJson.size());
+    }
```
1 тест - 1 запрос. Если тестируется удаление, то по http нужно отправить запрос только на удаление, а результат уже проверять через таск менеджер. По остальным тестам аналогично

Ivanov-s-tmn commented on test/server/HttpMethodDeleteTest.java:
```diff
@@ -0,0 +1,180 @@
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+public class HttpMethodDeleteTest extends HttpTaskServerTest {
```
Тесты лучше разделить по хендлерам, а не по методам

puffpundb committed:
Изменена логика менеджера на сервере.
Удалены ненужные методы из сервера.
Удалены лишние эндпоинты истории.
Изменены тесты.
Тесты разделены по хендлерам.
